### PR TITLE
Docker Compose profiles + Dredd

### LINKED ISSUE #1635: Run All Supported Versions and Check Hashes before Release
to avoid problems like #1309

### LINKED ISSUE #1739: Test Idempotent API Endpoints against Node in CI, using Example Calls in Documentation
Check performance when we do—make sure latency is below a certain threshold.

@ouziel-slama this seem reasonable?

### PR DESCRIPTION
**Pull request by ouziel-slama**
- [x] support `testnet` and `mainnet` on the same computer
- [x] update documentation
- [x] test docker compose on `testnet` and `mainnet`
- [x] run `dredd` (fix #1739)
- [x] run Compare Hashes test (fix #1635)

I struggled for several hours with the YAML features of docker compose and ultimately I couldn't find a simpler way to avoid having to define 4 or 5 environment variables:
```
docker compose --profile mainnet up -d
docker compose --profile testnet up -d
```
`--profile` is required for mainnet :-/

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,55 +49,102 @@ services:
...
       - "--api-host=0.0.0.0"
       - "--catch-up=bootstrap"
       - "--enable-api-v1"
       #- "--verbose"
-    environment:
-      - "XDG_DATA_HOME=/data/"
-      - "XDG_LOG_HOME=/data/"
 
+  ###################
+  # Tesnet Services #
```
Comment (adamkrellenstein) [inline docker-compose.yml:88]: Typo :)

**Review by adamkrellenstein — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (ouziel-slama) [reply docker-compose.yml]: thanks! fixed.

**Comment by ouziel-slama** [comment]
comes with https://github.com/CounterpartyXCP/Documentation/pull/194/files

**Comment by ouziel-slama** [comment]
ok so the new Docker Compose Test:
- also test `mainnet`
- run `dredd`
- run Compare Hashes Test (for now with api1.counterparty.io and api3.counterparty.io)

**Pull request merged**